### PR TITLE
Add tsc task and fix most compile errors

### LINKED ISSUE #15: Typescript compile has errors (using the tsc version in the starter kit)
Nice work on the starter kit!

I add a npm script to run the typscript compiler in `package.json` like this:

``` json
{
  "scripts": {
    "tsc": "tsc -p src",
  }
}
```

Then when I run `npm run tsc` I get a bunch of compile errors. 
There are some things missing in `tsconfig.json`. I'll try to do a PR to show more what I mean.

### PR DESCRIPTION
**Pull request by jonaskello**
Fixes most problems in #15

### CODE REVIEW

File: src/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "compileOnSave": false,
   "compilerOptions": {
+    "skipLibCheck": true,
```
Comment (piotrwitek) [inline src/tsconfig.json]: lib check is mandatory

**Comment by jonaskello** [comment]
Agreed on skipLibCheck. I set it to false. I also did some more changes in tsconfig and now all compile errors but one is fixed. Specifically I had to change noResolve to false and module to commonjs. I did this by trial and error so not sure why.

The remaining error is this:

```
src/containers/currency-converter-container/components/currency-valuation-header                            -calendar.tsx(2,22): error TS2307: Cannot find module 'react-input-calendar'.
```

I think it might be because there are no types installed for react-input-calendar?

**Comment by piotrwitek** [comment]
noResolve makes sense, because it strip out /// references to definition which are used in @types :)
IMO module should be system, because it will validate all various import types not only require
types for react-input-calendar should not be used, because the file actually importing that is not used by main app module, should check and fix includes in tsconfig

File: src/tsconfig.json [hunk 2]
```diff
@@ -1,17 +1,18 @@
 {
   "compileOnSave": false,
   "compilerOptions": {
+    "skipLibCheck": false,
     "declaration": false,
     "jsx": "react",
-    "module": "es6",
+    "module": "commonjs",
```
Comment (piotrwitek) [inline src/tsconfig.json]: please remove this change

File: src/tsconfig.json [hunk 3]
```diff
@@ -30,7 +31,7 @@
     "allowJs": false,
     "allowSyntheticDefaultImports": true,
     "noImplicitUseStrict": false,
-    "lib": ["es5", "es6", "es2016.array.include"],
+    "lib": ["es5", "es6", "es2016.array.include", "dom", "es2016.array.include", "es2015.promise", "es2017.object", "es2015.iterable"],
```
Comment (piotrwitek) [inline src/tsconfig.json]: please remove doubled es2016.array.include

**Review by piotrwitek — CHANGES_REQUESTED** [review]
With few additional fixes I have resolved this issue, please do requested changes and I'll merge this cr

**Comment by jonaskello** [comment]
I think the requested changes are in now :-)

**Review by piotrwitek — APPROVED** [review]

**Comment by piotrwitek** [comment]
@jonaskello merged, please check updated readme

**Pull request merged**